code/modules/surgery/organs/cybernetic_implants.dm

```diff
@@ -0,0 +1,236 @@
+	name = "thermals implant"
+	desc = "These cybernetic eyes will give you Thermal vision. Vertical slit pupil included."
+	eye_color = "FC0"
+	implant_color = "#FFCC00"
+	flash_protect = -1
+
+/obj/item/cybernetic_implant/eyes/thermals/function()
+	if(!owner)
+		return
+
+	owner.sight |= SEE_MOBS
+	owner.permanent_sight_flags |= SEE_MOBS
+	update_eye_color("You see prey everywhere you look...")
+
+/obj/item/cybernetic_implant/eyes/emp_act(severity)
+	if(severity > 1)
+		if(prob(5))
+			return
+	var/save_sight = owner.sight
+	owner.sight &= 0
+	owner.disabilities |= BLIND
+	owner << "<span class='warning'>Static obfuscates your vision!</span>"
+	spawn(50)
+	owner.sight |= save_sight
+	owner.disabilities &= ~BLIND
+
+
+//[[[[BRAIN]]]]
+
+/obj/item/cybernetic_implant/brain
+	name = "cybernetic brain implant"
+	desc = "injectors of extra sub-routines for the brain"
+	icon_state = "brain_implant"
+
+/obj/item/cybernetic_implant/brain/New()
+	var/icon/overlay = new /icon('icons/obj/surgery.dmi',"brain_implant_overlay")
+	overlay.ColorTone(implant_color)
+	overlays |= overlay
+	..()
+
+/obj/item/cybernetic_implant/brain/emp_act(severity)
+	var/stun_amount = 5 + (severity-1 ? 0 : 5)
+	owner.Stun(stun_amount)
+	owner << "<span class='warning'>Your body seizes up!</span>"
+	return stun_amount
+
+/obj/item/cybernetic_implant/brain/anti_drop
+	name = "anti-drop implant"
+	desc = "This cybernetic brain implant will allow you to force your hand muscles to contract, preventing item dropping. Twitch ear to toggle."
+	var/active = 0
+	var/l_hand_ignore = 0
+	var/r_hand_ignore = 0
+	var/obj/item/l_hand_obj = null
+	var/obj/item/r_hand_obj = null
+	implant_color = "#DE7E00"
+
+/obj/item/cybernetic_implant/brain/anti_drop/function()
+	action_button_name = "Toggle Anti-Drop"
+
+/obj/item/cybernetic_implant/brain/anti_drop/ui_action_click()
+	active = !active
+	if(active)
+		l_hand_obj = owner.l_hand
+		r_hand_obj = owner.r_hand
+		if(l_hand_obj)
+			if(owner.l_hand.flags & NODROP)
+				l_hand_ignore = 1
+			else
+				owner.l_hand.flags |= NODROP
+				l_hand_ignore = 0
+
+		if(r_hand_obj)
+			if(owner.r_hand.flags & NODROP)
+				r_hand_ignore = 1
+			else
+				owner.r_hand.flags |= NODROP
+				r_hand_ignore = 0
+
+		if(!l_hand_obj && !r_hand_obj)
+			owner << "<span class='notice'>You are not holding any items, your hands relax...</span>"
+			active = 0
+		else
+			var/msg = 0
+			msg += !l_hand_ignore && l_hand_obj ? 1 : 0
+			msg += !r_hand_ignore && r_hand_obj ? 2 : 0
+			switch(msg)
+				if(1)
+					owner << "<span class='notice'>Your left hand's grip tightens.</span>"
+				if(2)
+					owner << "<span class='notice'>Your right hand's grip tightens.</span>"
+				if(3)
+					owner << "<span class='notice'>Both of your hand's grips tighten.</span>"
+	else
+		if(!l_hand_ignore && owner.l_hand && owner.l_hand == l_hand_obj)
+			owner.l_hand.flags ^= NODROP
+		if(!r_hand_ignore && owner.r_hand && owner.r_hand == r_hand_obj)
+			owner.r_hand.flags ^= NODROP
```

Comment:
This could still cause people to have NODROP jumpsuits that they are wearing. I suggest doing this:
```
if(!r_hand_ignore && r_hand_obj in owner.contents)
    r_hand_obj.flags ^= NODROP
```
it's not perfect, but it's the best we can do.

(also make this a separate release_item() proc so you dont have to copypaste it in the emp_act())